services/src/main/resources/org/keycloak/protocol/oidc/endpoints/login-status-iframe.html

```diff
@@ -120,6 +124,42 @@
         }
         return null;
       }
+
+      /**
+       * @param {ArrayBuffer} bytes
+       * @see https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem
+       */
+      function bytesToBase64(bytes) {
+        const binString = String.fromCodePoint(...bytes);
+        return btoa(binString);
+      }
+
+      /**
+        * @param {string} message
+        * @see https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#basic_example
+        */
+      async function sha256Digest(message) {
+        const encoder = new TextEncoder();
+        const data = encoder.encode(message);
+
+        if (typeof crypto === "undefined" || typeof crypto.subtle === "undefined") {
```

Comment:
Note that the Web Crypto API might not be available in all deployments of Keycloak. We might have to load the `web-crypto-shim.js` on [insecure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) (e.g. a non local IP, or a host without HTTPS).

See:
https://github.com/keycloak/keycloak/blob/8855cf231690e2661f777d8f78214fe9c2826031/js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/index.ftl#L82-L84

Reply:
This is great, I'll resolve this thread 👍